proposals/nnnn-mutex.md

```diff
@@ -11,18 +11,11 @@ This proposal introduces a mutual exclusion lock, or a mutex, to the standard li
+Ensuring your mutable state is properly synchronized in a coherent way may be as simple as changing your `class` type to an `actor`. However, there may be cases where it's not so simple to do so. For example, a historically synchronous code base may not be able to adopt features like `async`/`await` for any number of reasons, but couldn't embrace the actor model because of that restriction. Perhaps there is data that is allocated together, but isn't mutually exclusive with each other. Should you just allocate two actors?
 
-* [`_ManagedCriticalState` in the Observation module](https://github.com/apple/swift/blob/main/stdlib/public/Observation/Sources/Observation/Locking.swift)
-* [`LockedState` in swift-foundation](https://github.com/apple/swift-foundation/blob/main/Sources/_FoundationInternals/LockedState.swift)
-* [`NIOLock` in Swift NIO](https://github.com/apple/swift-nio/blob/main/Sources/NIOConcurrencyHelpers/NIOLock.swift)
-* [`OSAllocatedUnfairLock`](https://developer.apple.com/documentation/os/osallocatedunfairlock)
-
-Of course the main issue is that there isn't a single standardized implementation for this kind of type resulting in everyone essentially needing to roll their own. Rolling your own lock implementation isn't an easy endeavor. Many implementations either 1. use `UnsafeMutablePointer.allocate` or 2. go through `ManagedBuffer` both to get access to a stable address but to also ensure there won't be any intermediate copies or exclusivity checks. Both of those solutions are super inefficient because of the allocation which is completely unnecessary, but unfortunately it was the only solution Swift had to offer in this space. This is the core reason why the standard library has never proposed a proper mutex type because we always knew it would not be the solution we wanted.
-
-Until now the only workable solution to implement locking in Swift has involved explicit heap allocations, either manual or managed. We can do better.
+Whatever the reason may be, it may not be feasible to use an actor. In such cases, Swift doesn't really offer great solutions to help synchronize such access to mutable data. Many Swift programs opt to use ad-hoc implementations of a mutal exclusion lock, or a mutex. A mutex is a simple and easy to use synchronization primitive to help protect shared mutable data by ensuring that a single execution context has exclusive access to the related data. The main issue is that there isn't a single standardized implementation for this synchronization primitive resulting in everyone needing to roll their own.
```

Comment:
```suggestion
Whatever the reason may be, it may not be feasible to use an actor. In such cases, Swift currently is missing standardized tools to offer developers ensure proper synchronization in their concurrent data-structures. Many Swift programs opt to use ad-hoc implementations of a mutal exclusion lock, or a mutex. A mutex is a simple to use synchronization primitive to help protect shared mutable data by ensuring that a single execution context has exclusive access to the related data. The main issue is that there isn't a single standardized implementation for this synchronization primitive resulting in everyone needing to roll their own.
```

Driving more home that we're just not providing tools, rather than some great story about it. Plain wording is better here, doesn't sound like there's some amazing new invention coming up :)